orchagent/lagorch.cpp

```diff
@@ -0,0 +1,217 @@
+{
+    SWSS_LOG_ENTER();
+
+    if (consumer.m_toSync.empty())
+        return;
+
+    if (!m_portsOrch->isInitDone())
+        return;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        size_t found = key.find(':');
+        string lag_alias, port_alias;
+        if (found == string::npos)
+            lag_alias = key;
+        else
+        {
+            lag_alias = key.substr(0, found);
+            port_alias = key.substr(found+1);
+        }
+
+        string op = kfvOp(t);
+
+        /* Manipulate LAG when port_alias is empty */
+        if (port_alias == "")
+        {
+            if (op == SET_COMMAND)
+            {
+                /* Duplicate entry */
+                if (m_lags.find(lag_alias) != m_lags.end())
+                {
+                    SWSS_LOG_ERROR("Duplicate LAG entry alias:%s", lag_alias.c_str());
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+
+                if (addLag(lag_alias))
+                    it = consumer.m_toSync.erase(it);
+                else
+                    it++;
+            }
+            else if (op == DEL_COMMAND)
+            {
+                assert(m_lags.find(lag_alias) != m_lags.end());
+                Port lag;
+                assert(m_portsOrch->getPort(lag_alias, lag));
+
+                if (removeLag(lag))
+                    it = consumer.m_toSync.erase(it);
+                else
+                    it++;
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown operation type %s\n", op.c_str());
+                it = consumer.m_toSync.erase(it);
+            }
+        }
+        /* Manipulate member */
+        else
+        {
+            assert(m_lags.find(lag_alias) != m_lags.end());
+            Port lag, port;
+            assert(m_portsOrch->getPort(lag_alias, lag));
+            assert(m_portsOrch->getPort(port_alias, port));
+
+            if (op == SET_COMMAND)
+            {
+                /* Duplicate entry */
+                if (m_lags[lag_alias].find(port_alias) != m_lags[lag_alias].end())
+                {
+                    SWSS_LOG_ERROR("Duplicate LAG member entry lag:%s port:%s",
+                            lag_alias.c_str(), port_alias.c_str());
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+
+                assert(port.m_type == Port::PHY_PORT);
+
+                if (addLagMember(lag, port))
+                    it = consumer.m_toSync.erase(it);
+                else
+                    it++;
+            }
+            else if (op == DEL_COMMAND)
+            {
+                assert(m_lags[lag_alias].find(port_alias) != m_lags[lag_alias].end());
+                assert(port.m_type == Port::LAG_MEMBER);
+
+                if (removeLagMember(lag, port))
```

Comment:
check if the port is already a lag member before you do the remove, you need to perform this sanity check in at APP DB level.